SWIPs/swip-20-honey_to_money.md

```diff
@@ -0,0 +1,103 @@
+Swarm needs a mechanism to set prices which makes sense relative to other (distributed storage) services in the market. As the absolute price of using Swarm is defined by the cost of honey (Swarm's internal accounting unit), this SWIP proposes a minimal interface to enable upgrading honey prices, network-wide and for all nodes. The specifics on what an efficient price should reflect and how the update process looks like is *not* part of this SWIP.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+It is expected that the price of using Swarm will change over time to reflect the changing economic landscape as well as to facilitate experimentation. Such price updates must happen atomically to prevent accounting unbalances from appearing. To lay the groundwork for enabling such price changes, this SWIP proposes a minimal interface for a *price-oracle*. Furthermore, this SWIP includes a proposal for implementing an *on-chain* price oracle, governed by Swarm developers and stakeholders to allow for experimentation.
+ 
+This SWIP is part of a series of SWIPs (but can be implemented on its own). To see the full picture, please refer to [swip-message_to_honey](https://github.com/Eknir/SWIPs/blob/SWIP-message_to_honey/SWIPs/swip-message_to_honey.md), [swip-multiple_payment_processing_support](https://github.com/Eknir/SWIPs/blob/SWIP-multiple_payment_processing_support/SWIPs/swip-multiple_payment_processing_support.md) and the diagram below:
+![SWIP_Diagrams.png](./../assets/swip-honey_to_money/SWIP_Diagrams.png)
+ 
+## Motivation
+<!--The motivation is critical for SWIPs that want to change the Swarm protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SWIP solves. SWIP submissions without sufficient motivation may be rejected outright.-->
+Nodes keep track of balances relative to each other using an internal reference unit: honey. Since honey is not a currency in which nodes can settle their balances with each other—it is merely a unit of account—there is a requirement to facilitate converting honey to a currency. Furthermore, it is desirable that this conversion rate can fluctuate to reflect the changing economic landscape and facilitate experimentation.
+
+This update process should be atomic: either all nodes upgrade or none of them do. The current implementation hardcodes honey prices in the source code. This violates the atomicity requirement as there is no guarantee that all Swarm nodes are on the same release. By externalizing the price to an oracle, Swarm completely decouples its core functions from price discovery. By building support for price oracles, it will be possible to coordinate price changes across the network and set the stage for future experimentation with various mechanisms for setting prices, from fixed prices to DAOs or to bonding curves.
```

Comment:
> By externalizing the price to an oracle, Swarm completely decouples its core functions from price discovery.

The choice is not binary between and oracle and having a hardcoded number in the implementation. I don't see a coherent argument here presented for why an oracle is the only way to reach consensus on per-message cost in terms of unit of account.

Reply:
This is described in the rationale sector, but to be honest with you: I am myself also not fully convinced of this being the way-to-go. 
To be clear: this is not about a _per-message cost_, but it is about the price you pay for resetting the balance with your peer back to zero, given an imbalance. 

If you welcome the idea, I would be happy to discuss with you the current rationale behind chosing the oracle and you can either help me to sharpen this rationale, or bash it completely (in which case we have to come up with a different solution). 

FYI: I have asked a professor in micro-economics/auction theory/game design to help me brainstorming on exactly this topic, so I would welcome to speak about it with you as well.

Reply:
I'm sorry, but I feel to do justice to a deeper discussion on the topic requires a certain degree of immersion, and I don't think I have enough time to set aside for that. 

As an audience, however, I think what I feel is missing at a minimum some reference to theoreical sources to back up the claims that are being made in this SWIP.

Reply:
Thank @nolash , this will be addressed.